src/patternfly/components/Panel/panel.scss

```diff
@@ -107,6 +110,13 @@
   padding-block-end: var(--#{$panel}__main-body--PaddingBlockEnd);
   padding-inline-start: var(--#{$panel}__main-body--PaddingInlineStart);
   padding-inline-end: var(--#{$panel}__main-body--PaddingInlineEnd);
+
+  &:has(.#{$panel}__menu) {
```

Comment:
I wonder if we might handle this a little different for a couple of reasons
* there is a [potential performance concern around using `:has()` ](https://github.com/patternfly/patternfly/issues/6694)
* in other components, the structure is usually something like `component__block` with `component__block-body` as a child (or children if you have multiple sections). So I'd probably expect `__menu` to be adjacent to `__main-body`

Since the react component is composable, wdyt about updating this so `__menu` is adjacent to `__main-body`? So in react it would look something like below, though I'm guessing if you're using this as a menu, you probably wouldn't use `__main-body`, you'd just have `__menu` in the main area?

```
  <Panel>
    <PanelHeader>Header content</PanelHeader>
    <Divider />
    <PanelMain tabIndex={0}>
      <PanelMainBody>
        a sentence or two
      </PanelMainBody>
      <PanelMenu>
        /* menu component */
      </PanelMenu>
      <PanelMainBody>
        a sentence or two
      </PanelMainBody>
    </PanelMain>
    <PanelFooter>Footer content</PanelFooter>
  </Panel>
```

Reply:
Updated the demo to just remov ethe MainBody, but the panel menu is still rendered inside the PanelMain component